### PR TITLE
WEBUMENIA-1901-komponent-select-pre-zoradenie-order-by-

### PR DESCRIPTION
**Pull request by FrantisekMichalSebestyen**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (link to Jira or GitHub issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [ ] I have updated the [CHANGELOG](../CHANGELOG.md)
- [ ] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: resources/js/components/catalog-new/Filter.vue [hunk 1]
```diff
@@ -20,6 +20,9 @@ export default {
         }
     },
     computed: {
+        sort() {
+            return this.getSort()
+        },
```
Comment (eronisko) [inline resources/js/components/catalog-new/Filter.vue:25]: No need for a computed property _and_ a method -- I'd suggest moving the contents of `getSort` here

File: resources/js/components/catalog-new/Filter.vue [hunk 2]
```diff
@@ -52,7 +55,9 @@ export default {
             }
         },
         lengthOfLabelItems() {
-            return Object.values(this.filters).filter((filterName) => filterName.list.some((item) => item.checked)).length
+            return Object.values(this.filters).filter((filterName) =>
+                filterName.list.some((item) => item.checked)
+            ).length
```
Comment (eronisko) [inline resources/js/components/catalog-new/Filter.vue:60]: Again, pls extract from URL -- looks like we'll need a `selectedValues` computed property

File: resources/js/components/catalog-new/Filter.vue [hunk 3]
```diff
@@ -64,6 +69,10 @@ export default {
             const urlQuery = this.$route.query
             return urlQuery[checkboxName]
         },
+        getSort() {
+            const urlQuery = this.$route.query
+            return urlQuery['sort'] || 'updated_at'
```
Comment (eronisko) [inline resources/js/components/catalog-new/Filter.vue:74]: It's fine to leave `sort` null/empty -- usually we'd handle default sorting on the back-end anyway. No need for a default on the front-end

File: resources/js/components/catalog-new/Filter.vue [hunk 4]
```diff
@@ -82,10 +91,22 @@ export default {
         setOpenedFilter(name) {
             this.openedFilter = name
         },
+        handleSort(sortValue) {
```
Comment (eronisko) [inline resources/js/components/catalog-new/Filter.vue:94]:
```suggestion
        handleSortChange(sortValue) {
```

File: resources/js/components/catalog-new/NewSort.vue [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                },
+                {
+                    value: 'view_count',
+                    text: 'počtu videní',
+                },
+                {
+                    value: 'random',
+                    text: 'náhodného poradia',
+                },
+            ]
```
Comment (eronisko) [inline resources/js/components/catalog-new/NewSort.vue:77]: Pls pass these as props (can be a static object for now no prob)

**Review by eronisko — CHANGES_REQUESTED** [review]

File: resources/js/components/catalog-new/Filter.vue (hunk 2 above)
Comment (FrantisekMichalSebestyen) [reply resources/js/components/catalog-new/Filter.vue:60]: Fixed in previous PR

**Review by eronisko — APPROVED** [review]
LGTM, thanks :)

**Pull request merged**